src/main/java/org/jenkinsci/plugins/workflow/cps/view/InterpolatedSecretsAction.java

```diff
@@ -0,0 +1,163 @@
+    }
+
+    public String getUrlName() {
+        return null;
+    }
+
+    public void record(@Nonnull String stepName, @Nonnull List<String> interpolatedVariables, @Nonnull String nodeId) {
+        interpolatedWarnings.add(new InterpolatedWarnings(stepName, interpolatedVariables, run, nodeId));
+    }
+
+    public List<InterpolatedWarnings> getWarnings() {
+       return interpolatedWarnings;
+    }
+
+    public boolean hasWarnings() {
+        if (interpolatedWarnings.isEmpty()) {
+            return false;
+        } else {
+            return true;
+        }
+    }
+
+    public boolean isInProgress() {
+        return run.isBuilding();
+    }
+
+    @Override
+    public void onAttached(Run<?, ?> run) {
+        this.run = run;
+    }
+
+    @Override
+    public void onLoad(Run<?, ?> run) {
+        this.run = run;
+    }
+
+    @ExportedBean
+    public static class InterpolatedWarnings {
+        final String stepName;
+        final List<String> interpolatedVariables;
+        final Run run;
+        final String nodeId;
+
+        private InterpolatedWarnings(@Nonnull String stepName, @Nonnull List<String> interpolatedVariables, @Nonnull Run run, @Nonnull String nodeId) {
+            this.stepName = stepName;
+            this.interpolatedVariables = interpolatedVariables;
+            this.run = run;
+            this.nodeId = nodeId;
+        }
+
+        @Exported
+        public String getStepSignature() {
+            StringBuilder sb = new StringBuilder();
+            Map<String, Object> stepArguments;
+            try {
+                stepArguments = getStepArguments(run, nodeId);
+            } catch (IllegalStateException e) {
+                return "Unable to construct " +  stepName + ": " + e.getMessage();
+            }
+
+            sb.append(stepName + "(");
+            Set<Map.Entry<String, Object>> entrySet = stepArguments.entrySet();
+            if (!entrySet.isEmpty()) {
+                boolean first = true;
+                for (Map.Entry<String, Object> argEntry : stepArguments.entrySet()) {
+                    Object value = argEntry.getValue();
+                    String valueString = String.valueOf(value);
+                    if (value instanceof ArgumentsAction.NotStoredReason) {
+                        switch ((ArgumentsAction.NotStoredReason) value) {
+                            case OVERSIZE_VALUE:
+                                valueString = "argument omitted due to length";
+                                break;
+                            case UNSERIALIZABLE:
+                                valueString = "unable to serialize argument";
+                                break;
+                            default:
+                                break;
+                        }
+                    }
+                    if (first) {
+                        first = false;
+                    } else {
+                        sb.append(", ");
+                    }
```

Comment:
To avoid needing this kind of logic and the `StringBuilder`, if you extract out an `argumentToString` method like I mentioned above, I think you could refactor most of this method into something more straightforward:

```
stepArguments.entrySet().stream()
        .map(InterpolatedSecretsAction::argumentToString)
        .collect(Collectors.joining(",", "(", ")"))
```

Reply:
done in bd0dfd285dee8361e9aceaaa2bc87ff1e2b6059e